### PR TITLE
nixos-rebuild: Add missing passed-through options to man page and completion

### LINKED ISSUE #16770: nixos-rebuild documentation doesn't mention all CLI options
## Issue description

The nixos-rebuild script has some undocumented options, such as "--repair" (does not appear in the man page). The actual flags can be viewed with `$EDITOR $(command -v nixos-rebuild)`.
## Technical details
- System: NixOS 16.03 and 16.09(unstable)

### PR DESCRIPTION
**Pull request by samueldr**
## Description of changes

This is syncing the man page and bash completion with the options. This fixes #16770.

In addition it adds the `repl` command to the bash completion.

I'm also adding myself to CODEOWNERS to ensure the bash completion doesn't drift again.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/nixos-rebuild/_nixos-rebuild [hunk 1]
```diff
@@ -71,7 +72,14 @@ _nixos-rebuild() {
...
     --max-jobs -j # number
     --log-format # format
+    --quiet
+    --print-build-logs -L
+    --no-build-output -Q
+    --fallback
+    --refresh
+    --repair
     -I # NIX_PATH
+    --offline # --no-net not suggested, deprecated
```
Comment (rjpcasalino) [inline pkgs/os-specific/linux/nixos-rebuild/_nixos-rebuild:82]: `--no=net` is preferred over `offline` nowadays?

Comment (samueldr) [reply pkgs/os-specific/linux/nixos-rebuild/_nixos-rebuild:82]: Hm? I don't understand.

`--no-net` is a now deprecated `nix` argument. So I'm not adding it to the bash completion. Does that help?

**Review by samrose — COMMENTED** [review]
Github says there is an error in CODEOWNERS related to jonringer being listed there, but being an "unkown owner"

I didn't know if that would impact merging this change, but thought I'd mention it

File: pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh [hunk 2]
```diff
@@ -81,7 +81,11 @@ while [ "$#" -gt 0 ]; do
         j="$1"; shift 1
         extraBuildFlags+=("$i" "$j")
         ;;
-      --accept-flake-config|-j*|--quiet|--print-build-logs|-L|--no-build-output|-Q| --show-trace|--keep-going|-k|--keep-failed|-K|--fallback|--refresh|--repair|--impure|--offline|--no-net)
+      --accept-flake-config|-j*|--quiet|--print-build-logs|-L|--no-build-output|-Q| --show-trace|--keep-going|-k|--keep-failed|-K|--fallback|--refresh|--repair|--impure|--offline)
+        extraBuildFlags+=("$i")
+        ;;
+      # Deprecated passed-through arguments
+      --no-net) # To be removed when removed from Nix; --offline is the replacement
```
Comment (rjpcasalino) [inline pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh:88]: oh `offline` is the replacement. OK, makes sense.

File: pkgs/os-specific/linux/nixos-rebuild/_nixos-rebuild (hunk 1 above)
Comment (rjpcasalino) [reply pkgs/os-specific/linux/nixos-rebuild/_nixos-rebuild:82]: ah saw your comment below sorry for noise

**Comment by samueldr** [comment]
> Github says there is an error in CODEOWNERS related to jonringer being listed there, but being an "unkown owner"
> 
> I didn't know if that would impact merging this change, but thought I'd mention it

The same errors are in `master`: https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS

Though since the ban is temporary, not an issue.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request closed without merging**